pkg/sql/row/writer.go

```diff
@@ -150,11 +152,24 @@ func prepareInsertOrUpdateBatch(
 				return nil, err
 			}
 
+			var oldVal []byte
```

Comment:
perhaps leave a todo around plumbing an `oldValueBuf` that we can use here and for hydrating `expBytes` below across `prepareInsertOrUpdateBatch` calls?

Reply:
I left a todo about allocations in general. I'm not sure that just plumbling this value buf would do that much given that presumably most of the allocation comes inside the Marshal functions.